File: generator-common/src/main/java/io/github/potjerodekool/openapi/common/Project.java
package io.github.potjerodekool.openapi.common;

import io.github.potjerodekool.openapi.common.dependency.DependencyChecker;

import java.nio.file.Path;
import java.util.List;

public record Project(Path rootDir,
                      List<Path> sourceRoots,
                      List<Path> resourcePaths,
                      Path generatedSourcesDirectory,
                      DependencyChecker dependencyChecker) {

}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/Project.kt
package io.github.potjerodekool.openapi.common

import io.github.potjerodekool.openapi.common.dependency.DependencyChecker
import java.nio.file.Path

data class Project(
    val rootDir: Path,
    val sourceRoots: List<Path>,
    val resourcePaths: List<Path>,
    val generatedSourcesDirectory: Path,
    val dependencyChecker: DependencyChecker
)
